sdk/storage/azure-storage-blob/azure/storage/blob/_shared_access_signature.py

```diff
@@ -66,13 +66,14 @@ def generate_blob(self, container_name, blob_name, snapshot=None, version_id=Non
         :param str snapshot:
             The snapshot parameter is an opaque DateTime value that,
             when present, specifies the blob snapshot to grant permission.
-        :param BlobSasPermissions permission:
```

Comment:
why are we removing these?

Reply:
Just moving the type to the below `:type permission:` tag since I added an `or` to the type. It seems like the style we follow is, if the type has an `or` we have it declared with a separate `type` line.

NOTE: `str` has always been supported, just updated the docs to reflect it.